apps/s_client.c

```diff
@@ -74,6 +74,7 @@ static void print_stuff(BIO *berr, SSL *con, int full);
 static int ocsp_resp_cb(SSL *s, void *arg);
 #endif
 static int ldap_ExtendedResponse_parse(const char *buf, long rem);
+static char *base64encode (const void *buf, size_t len);
```

Comment:
There are already a couple of base64 encoding implementations in openssl.  It would probably be better to find one of them and use that (although it might be ugly and you might want a friendly wrapper) rather than write yet another one.

Reply:
Ignore me; I see you already did that right thing.